connectors/src/connectors/microsoft/index.ts

```diff
@@ -168,6 +172,17 @@ export class MicrosoftConnectorManager extends BaseConnectorManager<null> {
       );
     }
 
+    const isTablesView = viewType === "tables";
+    if (isTablesView && filterPermission !== "read") {
+      return new Err(
+        new Error("Tables view is only supported for read permissions")
+      );
+    }
+
+    if (isTablesView) {
+      return getFoldersAndSpreadSheet(connector, parentInternalId);
```

Comment:
@philipperolet thinking of doing the same for https://github.com/dust-tt/dust/pull/6321